lib/orderbook/OrderBook.ts

```diff
@@ -345,6 +352,53 @@ class OrderBook extends EventEmitter {
     return outgoingOrder;
   }
 
+  /**
+   * Handles a request from a peer to create a swap deal. Checks if the order for the requested swap
+   * is available and if a route exists to determine if the request should be accepted or rejected.
+   * Responds to the peer with a swap response packet containing either an accepted quantity or rejection reason.
+   */
+  private handleSwapRequest = async (requestPacket: SwapRequestPacket, peer: Peer)  => {
+    assert(requestPacket.body, 'SwapRequestPacket does not contain a body');
+    assert(this.swaps, 'swaps module is disabled');
+    let rejectionReason: SwapRejectionReason | undefined;
+    let acceptedQuantity = 0;
+    const requestBody = requestPacket.body!;
+
+    if (!this.matchingEngines.has(requestBody.pairId)) {
+      rejectionReason = SwapRejectionReason.PAIR_NOT_SUPPORTED;
```

Comment:
I think so, basically this is checking if we even have a matching queue for the `pairId` specified by the swap request. Technically there could be a a valid `orderId` along with a `pairId` that we don't support.

Reply:
how can there be an orderId (that we accepted) for a paid that we don't support?

Reply:
It wouldn't be a valid request from the taker, that's for sure. It might make sense to penalize the taker for this, I can put in a todo. It's an unexpected edge case, but I think it's something we need to check since if the `pairId` is invalid then we won't even have a collection in which to check for the `orderId`.